src/property.hpp

```diff
@@ -204,6 +224,14 @@ inline Property::Property(std::string name, PropertyType type,
 {
 }
 
+inline Property::Property(std::string name, PropertyType type,
```

Comment:
This constructor is redundant.

Reply:
It is being used by tests and not having it results in:
```
Undefined symbols for architecture x86_64:
  "realm::Property::Property(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, realm::PropertyType, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)", referenced from:
      ____C_A_T_C_H____T_E_S_T____0() in list.cpp.o
      ____C_A_T_C_H____T_E_S_T____3() in migrations.cpp.o
      ____C_A_T_C_H____T_E_S_T____111() in migrations.cpp.o
      ____C_A_T_C_H____T_E_S_T____134() in migrations.cpp.o
      ____C_A_T_C_H____T_E_S_T____0() in object.cpp.o
      ____C_A_T_C_H____T_E_S_T____3() in object_store.cpp.o
      ____C_A_T_C_H____T_E_S_T____102() in results.cpp.o
```

Reply:
It is functionally identical to the three-argument version with a default value for the third argument (i.e. what previously existed).

Reply:
Those was changed as well, these constructors have no default values anymore as having two 3-argument constructors with different defaults is ambiguous. So there is now a constructor for creating backlinks (with a string default value) and one for creating normal links (with a relationship default value)

Reply:
Having `Property(std::string name, PropertyType type, std::string object_type, std::string link_origin_property_name="");` and `Property(std::string name, PropertyType type, std::string object_type, Relationship relationship);` is not ambiguous.

Reply:
Ah, like that. No, your right.

Reply:
2-arg constructor has been removed